### ISSUE TITLE
Implement (semantically void) 'exports', now

### ISSUE THREAD

**Issue by masak** [masak/007#404]
Since @claes-magnus basically started writing modules, we should have `exports` early, even before we have the module system, so that we can write the module files as correctly as possible and reduce the impedance mismatch when modules become a thing.

Simplest I can think of (which should be enough for now) is to simple slap an optional `exports` in front of some rules. `func`, `macro`, `my` and `class` should be enough for now.

Of course, some tests for this would be in order too. But we don't need to test for any semantics yet.

**Comment by masak** [comment]
@claes-magnus In the case of `007-ascii-header-printer`, it'd mean [this line](https://github.com/claes-magnus/007-ascii-header-printer/blob/master/ascii_header_printer.007#L252) could instead be written as

    export func printHeader(string) {

And it wouldn't make any difference _now_, but it'd automatically make the project play well once we have #53 implemented.

**Comment by masak** [comment]
> `my`

Ahahah. But since #369, `my` is no longer a statement.

So,

1. Should we allow `export` on _all_ `my` terms, even if they are somewhere far in the middle of some statement?

2. Should we accept `export` but only at the beginning/top of an expression statement?

3. Should we back out of `my` being a term?

Leaning towards "most likely not", "probably", and "no" on these for now. But it's an unexpected complication, so I will mull over it some.

(`export` and `@constant` are now two things that sorta-kinda expect there to be a `Q.Statement.My`, but that will probably have to live with there not being one.)

**Comment by masak** [comment]
Going with option 2 for now. This is the "we pretend `my` is a statement sometimes" option.

For that to work, we need to have an algorithm to figure out what "at the beginning" of an expression statement means. I wrote "beginning/top", but it's actually not always at the top.

As far as I can see, this should work for finding a possible `my` at the beginning of an expression:

* Is the top node a `my` term? Then we found it.
* Is the top node a prefix or a term? Then there's no `my` term at the beginning.
* Is the top node a postfix? Then recurse into its term and start from the top of this list.
* Is the top node an infix? Then recurse into its left-hand side and start from the top of this list.

I think this should work out even with custom operators and whatnot.
